synth/src/datasource/postgres_datasource.rs

```diff
@@ -273,17 +273,50 @@ impl RelationalDataSource for PostgresDataSource {
                 begin: None,
                 end: None,
             }),
+            "time" => Content::DateTime(DateTimeContent {
+                format: "%H:%M:%S".to_string(),
+                type_: ChronoValueType::NaiveTime,
+                begin: None,
+                end: None,
+            }),
+            "json" => Content::Object(ObjectContent {
+                skip_when_null: false,
+                fields: BTreeMap::new(),
+            }),
             "uuid" => Content::String(StringContent::Uuid(Uuid)),
-            _ => bail!("We haven't implemented a converter for {}", data_type),
+            _ => {
+                if let Some(inner) = data_type.strip_prefix('_') {
+                    Content::Array(ArrayContent::from_content_default_length(
+                        self.decode_to_content(inner, char_max_len)?,
+                    ))
+                } else {
+                    bail!("We haven't implemented a converter for {}", data_type)
+                }
+            }
         };
 
         Ok(content)
     }
 
-    fn extend_parameterised_query(query: &mut String, curr_index: usize, extend: usize) {
+    fn extend_parameterised_query(query: &mut String, curr_index: usize, query_params: Vec<Value>) {
+        let extend = query_params.len();
+
         query.push('(');
-        for i in 0..extend {
-            query.push_str(&format!("${}", curr_index + i + 1));
+        for (i, param) in query_params.iter().enumerate().take(extend) {
```

Comment:
I think the `.take` call here is unnecessary since you're just parsing the length of `query_params`?